source/exe/main.cc

```diff
@@ -20,6 +21,12 @@
 #include "ares.h"
 #include "spdlog/spdlog.h"
 
+#if __cplusplus < 201103L ||                                                                       \
```

Comment:
Sure, that's a great idea.  What do you think of source/common/common as the location for that file?